File: 0057-insert-interval/0057-insert-interval.java
class Solution {
    public int[][] insert(int[][] intervals, int[] newInterval) {
        Queue<Integer> q = new LinkedList<>();

        for(int i=0; i<intervals.length; i++){
            if(newInterval[1] < intervals[i][0]){
                q.offer(newInterval[0]);
                q.offer(newInterval[1]);

                for(int j=i; j<intervals.length; j++){
                    q.offer(intervals[j][0]);
                    q.offer(intervals[j][1]);
                }
                
                System.out.println(q);
                int[][] res = new int[q.size()/2][2];
                int k = 0;

                while(!q.isEmpty()){
                    res[k][0] = q.poll();
                    res[k++][1] = q.poll();
                }

                return res;
            }
            else if(newInterval[0] > intervals[i][1]){
                q.offer(intervals[i][0]);
                q.offer(intervals[i][1]);
            }
            else{
                newInterval[0] = Math.min(newInterval[0], intervals[i][0]);
                newInterval[1] = Math.max(newInterval[1], intervals[i][1]);
            }
        }

        q.offer(newInterval[0]);
        q.offer(newInterval[1]);

        int[][] res = new int[q.size()/2][2];
        int k = 0;
        while(!q.isEmpty()){
            res[k][0] = q.poll();
            res[k++][1] = q.poll();
        }

        return res;
    }
}

File: 0057-insert-interval/0057-insert-interval.py
class Solution(object):
    def insert(self, intervals, newInterval):
        """
        :type intervals: List[List[int]]
        :type newInterval: List[int]
        :rtype: List[List[int]]
        """
        n = len(intervals)
        res = []
        i = 0

        #Not overlapping in starting 
        while i<n and newInterval[0] > intervals[i][1]:
            res.append(intervals[i])
            i+=1

        #string when overlap
        while i<n and newInterval[1] >= intervals[i][0]:
            newInterval[0] = min(newInterval[0], intervals[i][0])
            newInterval[1] = max(newInterval[1], intervals[i][1])
            i+=1

        #Insert new interval 
        res.append(newInterval)

        #add left overs 
        while i<n:
            res.append(intervals[i])
            i+=1

        return res